packages/web3-providers-ipc/src/index.ts

```diff
@@ -110,6 +110,23 @@ export default class IpcProvider<
 		this._removeSocketListeners();
 		this._addSocketListeners();
 	}
+	public async waitForConnection(): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			const maxNumberOfAttempts = 10;
+			const timeOutTime = 5000; // ms
+			let currentAttempt = 0;
```

Comment:
Instead of only hard coding these, I think there should be defaults and these should be configurable via setters at IpcProvider class level.